### PR TITLE
Reduce Qt margins, tweak scrollbar and splitters

### PR DESCRIPTION
**Pull request by kleinerpirat**
Good morning! No worries, this should be easy to approve :-)

# Remove Qt margin from editor webview
Anyone who's ever changed the background of the editor knows that the webview doesn't span the full width of the window due to Qt margins:

image

Also, because `overflow` is set to `auto` on most scrollable areas, the width changes when content overflows.

I removed the margins from the editor `.ui` files and replaced occurrences of `overflow: auto` with `overflow: scroll`. This is the result:

image

# (outdated, see comment below) Make custom scrollbar slimmer

Trying out Anki's webviews on mobile yesterday, I was stunned how annoying scrollbars can be on a slim aspect ratio. To improve the situation, I made the custom scrollbar *look* slimmer both in `ts` and Qt. The clickable area remains the same.

Inspired by GitHub Desktop:

image

### QScrollBar

---
Due to the scrollbar + margin changes, we got more horizontal space now, especially in the browser.

## Other changes
- remove icon from `QSplitter`, making it much slimmer
- fix scrollbar appearing somewhere in the middle of the Import CSV screen
- move overflow from `Fields` to its wrapper `PaneContent`, enabling the scroll-shadows that have already been implemented

### CODE REVIEW

**Comment by kleinerpirat** [comment]
I played around with the browser splitters a bit.

Currently the splitters are 1px thick and invisible, so they need to be placed right at the edge between two widgets to be discoverable. For example, the edge between the browser table and the editor is an obvious border, just like edge between the sidebar and table.

Unfortunately, the scrollbar of the browser table is located _outside_ of it, so the splitter is offset to the right. This makes it unintuitive to use:

image


Today I found a way to move the scrollbar inside of the table. This makes the splitter position clear:

image

Because the slim scrollbar looks a bit ugly inside the table due to the small margins around it, I reverted a few of the changes and just decreased its overall width by 2px.

File: qt/aqt/browser/table/table.py [hunk 1]
```diff
@@ -55,6 +55,15 @@ def __init__(self, browser: aqt.browser.Browser) -> None:
...
         self._view = view
+        # move scrollbar inside table by adding it into the same grid cell
+        self._view.setVerticalScrollBarPolicy(Qt.ScrollBarPolicy.ScrollBarAlwaysOff)
+        self._scrollbarLyout = QVBoxLayout()
+        # make sure scrollbar doesn't overlap with table header
+        self._scrollbarLyout.addSpacing(28)
+        self._scrollbarLyout.addWidget(self._view.verticalScrollBar())
+        self.browser.form.tableGridLayout.addLayout(
+            self._scrollbarLyout, 0, 0, Qt.AlignmentFlag.AlignRight
+        )
```
[the remark this answers is not in the document]
Comment (kleinerpirat) [reply qt/aqt/browser/table/table.py]: I've added a check for `is_mac or mw.pm.force_custom_styles()` and fixed the padding issue in the editor and other screens.

**Comment by kleinerpirat** [comment]
https://github.com/ankitects/anki/pull/2260/commits/cabace7effcadc947fd3ed405dc1a49e16cf7b1d fixes the Graphs header disappearing from view on scroll.

**Comment by dae** [comment]
Desktop issues:

Minor: truncated bottom border
2022-12-19T14:46:57,639928545+10:00

Minor: a pixel or two or spacing from the scrollbar would be good
2022-12-19T14:47:54,934314885+10:00

Page content appears to the right of top bar
2022-12-19T14:49:22,943201612+10:00

**Comment by dae** [comment]
On mobile, the graphs screen is still odd - there appear to be two separate containers that are scrolling, leading to the page not scrolling all the way down on one swipe, and the top bar disappearing as the outer container is scrolled.

IMG_B47872B43007-1

Also a minor issue there - the extra padding on the right (presumably for the desktop scrollbar width) leads to an unbalanced appear. Issue may exist on macOS as well?

**Comment by dae** [comment]
2.1.57 will be another quick update; pushing this one back so it can have more time in testing/any kinks can be worked out.

**Comment by dae** [comment]
Just an update here - I'm nervous about this change, and have some doubts about whether this is the right way forward I'm afraid :-(. Assuming the double scrollbar issues mentioned above can be fixed, I'm also worried about other possible issues this might have on mobile platforms, such as [elastic scrolling](https://forums.ankiweb.net/t/i-suggest-reverting-scroll-changes-in-ankimobile-beta-20088-3/25407), and the way content flows under the top and bottom bars - the page extends to the edges of the screen, but with margins so that content starts out below the bar and not hidden by it.

**Comment by kleinerpirat** [comment]
I hear you. I will revise this PR and eliminate some of the more risky changes when I find the time. Got two important exams coming up, but I can get to work next week.

**Comment by dae** [comment]
No problem, all the best for the exams :-)

Do you have any preferences on the timing of the .57 release? If there's nothing you think must be done before a release, I was thinking of putting together an rc in the next day or two, with the intention of a stable release in about a week if that suits?

**Comment by kleinerpirat** [comment]
I'll see if I can push some small fixes after the first exam is done tomorrow afternoon. But I'm in no rush regarding the bigger things (like this PR, field label API, background image support).

**Comment by dae** [comment]
Please don't feel any pressure for the small things either - I don't want to disrupt your exams, and a .58 can come out not long after if necessary.

**Comment by dae** [comment]
Cleaning up old PRs - please open a new one if you'd like to continue this.

**Pull request closed without merging**